
- don't export libs needed for DBD
- don't add -L%{_libdir} to --link-ld output

--- apr-util-1.2.2/build/dbd.m4.exports
+++ apr-util-1.2.2/build/dbd.m4
@@ -64,7 +64,6 @@
   dnl Since we have already done the AC_CHECK_LIB tests, if we have it, 
   dnl we know the library is there.
   if test "$apu_have_pgsql" = "1"; then
-    APR_ADDTO(APRUTIL_EXPORT_LIBS,[-lpq])
     APR_ADDTO(APRUTIL_LIBS,[-lpq])
   fi
 ])
@@ -115,7 +114,6 @@
   dnl Since we have already done the AC_CHECK_LIB tests, if we have it, 
   dnl we know the library is there.
   if test "$apu_have_mysql" = "1"; then
-    APR_ADDTO(APRUTIL_EXPORT_LIBS,[-lmysqlclient_r])
     APR_ADDTO(APRUTIL_LIBS,[-lmysqlclient_r])
   fi
 ])
@@ -152,7 +150,6 @@
   dnl Since we have already done the AC_CHECK_LIB tests, if we have it, 
   dnl we know the library is there.
   if test "$apu_have_sqlite3" = "1"; then
-    APR_ADDTO(APRUTIL_EXPORT_LIBS,[-lsqlite3])
     APR_ADDTO(APRUTIL_LIBS,[-lsqlite3])
   fi
 ])
@@ -189,7 +186,6 @@
   dnl Since we have already done the AC_CHECK_LIB tests, if we have it, 
   dnl we know the library is there.
   if test "$apu_have_sqlite2" = "1"; then
-    APR_ADDTO(APRUTIL_EXPORT_LIBS,[-lsqlite])
     APR_ADDTO(APRUTIL_LIBS,[-lsqlite])
   fi
 ])
--- apr-util-1.2.2/apu-config.in.exports
+++ apr-util-1.2.2/apu-config.in
@@ -147,8 +147,10 @@
     ;;
     --link-ld)
     if test "$location" = "installed"; then
-        ### avoid using -L if libdir is a "standard" location like /usr/lib
-        flags="$flags -L$libdir -l$APRUTIL_LIBNAME"
+        if test "$prefix" != "/usr"; then
+            flags="$flags -L${libdir}"
+        fi
+        flags="$flags -l$APRUTIL_LIBNAME"
     else
         flags="$flags -L$APU_BUILD_DIR -l$APRUTIL_LIBNAME"
     fi
